Add CD & PD to concepts page (#1460)
fluxcd/website: content/en/flux/concepts.md

using our [Terraform Provider](https://github.com/fluxcd/terraform-provider-flux).

For more information, take a look at [the bootstrap documentation](installation.md#bootstrap).

## Continuous Delivery

Continuous Delivery is a software development practice that automates the software release 
process up to the point of deployment except that manual intervention is required at some point. 
As code changes are made, they undergo testing, and artifacts are subsequently created. 
These artifacts are deployed to a staging environment for additional manual testing and approval 

> before being released to production servers.

Author: Alright then

By adopting this practice, software teams can update their code more frequently and reliably with 
less risk, while maintaining control over deployment and improving overall software quality.

## Continuous Deployment

Continuous Deployment is a software development practice that streamlines the process of delivering 
code changes to production through automation.

As code changes are made, they are automatically built, tested, and deployed to production servers 
without human or manual intervention. This significantly reduces the time it takes to deliver software 
updates to users as the approach allows developers to focus on writing code, while automated processes 
handle the tasks of testing and deploying the code changes.

## Progressive Delivery

Progressive Delivery builds on Continuous Delivery by gradually rolling out new features or updates 
to a subset of users, allowing developers to test and monitor the new features in a controlled environment 
and make necessary adjustments before releasing them to everyone.

> Developers can use techniques like feature flags, canary releases, and A/B testing to minimize the chances 

Reviewer: Let's add links to https://glossary.cncf.io/canary-deployment/  and https://fluxcd.io/flagger/usage/deployment-strategies/

For more information, take a look at [the bootstrap documentation](installation.md#bootstrap).

## Continuous Delivery

> Continuous Delivery refers to the practice of delivering software updates frequently and reliably. 

Reviewer: ```suggestion
## Continuous Delivery

Continuous Delivery refers to the practice of delivering software updates frequently and reliably. 
```

For more information, take a look at continuous delivery as defined by the [CNCF](https://glossary.cncf.io/continuous-delivery/).

## Continuous Deployment

> Continuous Deployment is the practice of automatically deploying code changes to production once they have passed through automated testing. 

Reviewer: ```suggestion
## Continuous Deployment

Continuous Deployment is the practice of automatically deploying code changes to production once they have passed through automated testing. 
```

For more information, take a look at continuous deployment as defined by the [CNCF](https://glossary.cncf.io/continuous-delivery/).

## Progressive Delivery

Progressive Delivery builds on Continuous Delivery by gradually rolling out new features or updates to a subset of users, allowing developers to test and monitor the new features in a controlled environment and make necessary adjustments before releasing them to everyone.

> Developers can use techniques like feature flags, canary releases, and A/B testing to minimize the chances of introducing bugs or errors that could harm users or interrupt business operations. These strategies enable a controlled and gradual rollout of new features, ensuring a smooth and successful release that enhances user trust and improves the overall user experience.

Reviewer: ```suggestion
Developers can use techniques like feature flags, [canary releases](https://glossary.cncf.io/canary-deployment/), and A/B testing to minimize the chances of introducing bugs or errors that could harm users or interrupt business operations. These strategies enable a controlled and gradual rollout of new features, ensuring a smooth and successful release that enhances user trust and improves the overall user experience.
```

> For more information, take a look at [canary deployment](https://glossary.cncf.io/canary-deployment/) and [deployment strategies](https://fluxcd.io/flagger/usage/deployment-strategies/) in Flux.

Reviewer: ```suggestion
The Flux project offers a specialised controller called [Flagger](https://github.com/fluxcd/flagger) that implements various progressive delivery techniques. For more information, take a look at [Flagger deployment strategies](https://fluxcd.io/flagger/usage/deployment-strategies/).
```
